### DISCUSSION TITLE
External CDN with next.js 15 server actions

### DISCUSSION THREAD

**Discussion by volodyad** [vercel/next.js#73951] [Help] [answered]
### Summary

I use next with cloudfront cdn by caching the pages and chunks itself

There is a server action which invoked in useEffect
in next14 server action id was usually same between rebuilds
in next15 it seems that action id change on every rebuild - it results in breaking cache on every rebuild with error  
[Error: Failed to find Server Action "". This request might be from an older or newer deployment. Original error: Cannot read properties of undefined (reading 'workers')]

It seems relate to https://nextjs.org/blog/next-15#enhanced-security-for-server-actions
But what about cdn? with such random id generation there is no chance to cache properly
From the perspective of cdn cache I would to have static action id, more than this maybe it would be good to have the ability to set this  id so it won't be change. I think server actions should be treated as api action with static path/id so, previous uncached version would be able to access it.

Is there any docs how this id was generated before?
Is there any option to disable id generation?
Do I understand correctly that server actions should not be used with external cdn?

**Comment by icyJoseph** [comment, marked as answer]
Don't you have to use this bit? https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions-and-mutations#overwriting-encryption-keys-advanced

> When self-hosting your Next.js application across multiple servers, each server instance may end up with a different encryption key, leading to potential inconsistencies.
> 
> To mitigate this, you can overwrite the encryption key using the process.env.NEXT_SERVER_ACTIONS_ENCRYPTION_KEY environment variable. Specifying this variable ensures that your encryption keys are persistent across builds, and all server instances use the same key.

**Reply by volodyad** [reply]
Thank you, it helped
